lms/djangoapps/instructor/views/api_v2.py

```diff
@@ -269,3 +281,58 @@ def get(self, request, course_id):
         tasks_data = [extract_task_features(task) for task in tasks]
         serializer = InstructorTaskListSerializer({'tasks': tasks_data})
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+@method_decorator(cache_control(no_cache=True, no_store=True, must_revalidate=True), name='dispatch')
+class ChangeDueDateView(APIView):
+    """
+    Grants a due date extension to a student for a particular unit.
+    this version works with a new payload that is JSON and more up to date.
+    """
+    permission_classes = (IsAuthenticated, permissions.InstructorPermission)
+    permission_name = permissions.GIVE_STUDENT_EXTENSION
+    serializer_class = BlockDueDateSerializerV2
+
+    def post(self, request, course_id):
+        """
+        Grants a due date extension to a learner for a particular unit.
+
+        params:
+            blockId (str): The URL related to the block that needs the due date update.
+            due_datetime (str): The new due date and time for the block.
+            email_or_username (str): The email or username of the learner whose access is being modified.
+        """
+        serializer_data = self.serializer_class(data=request.data)
+        if not serializer_data.is_valid():
+            return JsonResponseBadRequest({'error': _('All fields must be filled out')})
```

Comment:
```suggestion
            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```
Where Response comes from: **rest_framework.response import Response**

Reply:
serialize errors contains all the detail of that. That's why is better to define validations on the serializers.